node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -305,24 +304,38 @@ case class DataMessageHandler(
 
   private def sendNextGetCompactFilterHeadersCommand(
       peerMsgSender: PeerMessageSender,
-      stopHash: DoubleSha256DigestBE): Future[Boolean] =
+      prevStopHash: DoubleSha256DigestBE): Future[Boolean] =
     peerMsgSender.sendNextGetCompactFilterHeadersCommand(
       chainApi = chainApi,
       filterHeaderBatchSize = chainConfig.filterHeaderBatchSize,
-      stopHash = stopHash)
+      prevStopHash = prevStopHash)
 
   private def sendFirstGetCompactFilterHeadersCommand(
       peerMsgSender: PeerMessageSender): Future[Boolean] = {
+
     for {
       bestFilterHeaderOpt <-
         chainApi
           .getBestFilterHeader()
-      highestFilterBlockHash =
-        bestFilterHeaderOpt
-          .map(_.blockHashBE)
-          .getOrElse(DoubleSha256DigestBE.empty)
-      res <- sendNextGetCompactFilterHeadersCommand(peerMsgSender,
-                                                    highestFilterBlockHash)
+      blockHash = bestFilterHeaderOpt match {
+        case Some(filterHeaderDb) =>
+          filterHeaderDb.blockHashBE
+        case None =>
+          DoubleSha256DigestBE.empty
+      }
+      hashHeightOpt <- chainApi.nextBlockHeaderBatchRange(
+        prevStopHash = blockHash,
+        batchSize = chainConfig.filterHeaderBatchSize)
+      res <- hashHeightOpt match {
+        case Some((height, hash)) =>
+          peerMsgSender
+            .sendGetCompactFilterHeadersMessage(startHeight = height,
+                                                stopHash = hash)
+            .map(_ => true)
+        case None =>
+          sys.error(
```

Comment:
Maybe add to the error message that this likely means the chaindb has been corrupted or is in an invalid state